test/collection_test.js

```diff
@@ -64,6 +64,21 @@ describe("Collection", () => {
     };
   }
 
+  function createKeyListIdSchema() {
+    return {
+      generate(record) {
+        let id = [];
+        for (let key in record) {
+          id.push(key);
+        }
+        return id.join(",");
```

Comment:
Two things. First, this could be rewritten as:

```js
return Object.keys(record).join(",")
```

Also, we should ensure the keys are always ordered the same way to ensure consistency and predictability, so I'd suggest:

```js
return Object.keys(record).sort().join(",")
```